python/cudf/requirements.txt

```diff
@@ -0,0 +1,44 @@
+clang==8.0.1
```

Comment:
It looks like there's a bunch of things here that are for dev tooling, like clang, flake8, black, etc.

Can we trim this down to the minimal set of things needed to build and run cuDF from source?